src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PublicDataRetrievalService.java

```diff
@@ -0,0 +1,96 @@
+package uk.gov.companieshouse.overseasentitiesapi.service;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.model.officers.CompanyOfficerApi;
+import uk.gov.companieshouse.api.model.officers.OfficersApi;
+import uk.gov.companieshouse.overseasentitiesapi.client.ApiClientService;
+import uk.gov.companieshouse.overseasentitiesapi.exception.ServiceException;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+
+@Component
+public class PublicDataRetrievalService {
+    public static final String COMPANY_NUMBER = "company_number";
+    public static final String OFFICER_NAME = "officer_name";
+
+    private final ApiClientService apiClientService;
+
+    private CompanyProfileApi companyProfile;
+    private OfficersApi officers;
+
+    public PublicDataRetrievalService(ApiClientService apiClientService) {
+        this.apiClientService = apiClientService;
+    }
+    public OfficersApi getOfficerApi() {
+        return officers;
+    }
+    public CompanyProfileApi getCompanyProfile() {
+        return companyProfile;
+    }
+
+    public void initialisePublicData(String companyNumber, String passThroughTokenHeader) throws ServiceException {
+        this.companyProfile = getCompanyProfile(companyNumber, passThroughTokenHeader);
+        this.officers = getOfficers(companyNumber, passThroughTokenHeader);
+    }
+
+    private CompanyProfileApi getCompanyProfile(String companyNumber, String passThroughTokenHeader) throws ServiceException {
+        var logMap = new HashMap<String, Object>();
```

Comment:
you have not added the company name to the log map before you call info on line 44